### PR TITLE
Add test cases for Network.Socket and Inet.TCP modules

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Review by harendra-kumar — COMMENTED** [review]
Its a good idea to open separate PRs for networking and Unicode tests, since they are unrelated modules.

File: test/Streamly/Test/Unicode/Stream.hs [hunk 1]
```diff
@@ -59,6 +69,41 @@ propDecodeEncodeIdArrays =
...
+    forAll (choose (1, 1000)) $ \len ->
+        forAll (vectorOf len arbitraryASCIIChar) $ \list ->
+            monadicIO $ do
+                let wrds = SS.decodeLatin1 $ SS.encodeLatin1 $ S.fromList list
+                lst <- run $  S.toList wrds
+                assert (list == lst)
+
+propUnicodeToLatin1 :: Property
+propUnicodeToLatin1 =
+            monadicIO $ do
```
Comment (harendra-kumar) [inline test/Streamly/Test/Unicode/Stream.hs:89]: 4 space indent.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Please rebase and resolve the comments.

File: test/Streamly/Test/Network/Inet/TCP.hs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        & S.drain
+
+remoteAddr :: (Word8,Word8,Word8,Word8)
+remoteAddr = (127, 0, 0, 1)
+
+sender :: PortNumber -> MVar () -> SerialT IO Char
+sender port sem = do
+    _ <- liftIO $ takeMVar sem
+    liftIO $ threadDelay 1000000                       -- wait for server
+    S.replicate 1 testDataLarge                        -- SerialT IO String
```
[the remark this answers is not in the document]
Comment (rnjtranjan) [reply test/Streamly/Test/Network/Inet/TCP.hs]: Not sure about comment "S.yield?" Pls elaborate.

[the remark this answers is not in the document]
Comment (harendra-kumar) [reply test/Streamly/Test/Network/Inet/TCP.hs]: `S.yield` = `S.replicate 1`.

File: test/Streamly/Test/Network/Socket.hs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        & S.fold (SK.writeWithBufferOf 100 sk)
+        & discard
+
+handlerRW :: Socket -> IO ()
+handlerRW sk =
+          S.unfold SK.read sk
+        & S.fold (SK.write sk)
+        & discard
+------------------------------------------------------------------------------
+-- Accept connecttions and handle connected sockets
```
Comment (specdrake) [inline test/Streamly/Test/Network/Socket.hs:67]: there is a typo in `connections`

**Pull request merged**